portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -3505,6 +3505,25 @@ else if (groupId > 0) {
 					typeSettingsUnicodeProperties.getProperty("languageId");
 
 				if (Validator.isNotNull(defaultLanguageId)) {
+					HttpServletRequest originalHttpServletRequest =
+						getOriginalServletRequest(httpServletRequest);
+
+					String currentURL =
+						(String)originalHttpServletRequest.getAttribute(
+							WebKeys.CURRENT_URL);
+
+					if (Validator.isNotNull(currentURL) &&
+						currentURL.endsWith(".map")) {
+
+						if (initialize) {
+							setLocale(
```

Comment:
@locpham97 Some questions about the logic:
1. Why still call `setLocale` when the url is a `.map` if initialize?
2. This method is very long, so a lot of logic will be executed until it reaches here. I wonder if you can do this `.map` check earlier than here.

Reply:
I just revert to the original version I send PR. Moreover, I updated